applications/StructuralMechanicsApplication/test_examples/2DMPCCaseStrMesh.gid/run_test.py

```diff
@@ -0,0 +1,239 @@
+
+#### output settings start ####
+
+problem_path = os.getcwd()
+problem_name = ProjectParameters["problem_data"]["problem_name"].GetString()
+
+# initialize GiD  I/O (gid outputs, file_lists)
+from gid_output_process import GiDOutputProcess
+output_settings = ProjectParameters["output_configuration"]
+gid_output = GiDOutputProcess(main_model_part,
+                              problem_name,
+                              output_settings)
+
+gid_output.ExecuteInitialize()
+#### output settings end ####
+
+## Sets strategies, builders, linear solvers, schemes and solving info, and fills the buffer
+solver.Initialize()
+
+print(" ")
+print("::[KSM Simulation]:: Analysis -START- ")
+
+for process in list_of_processes:
+    process.ExecuteBeforeSolutionLoop()
+    
+## Set results when are written in a single file
+gid_output.ExecuteBeforeSolutionLoop()
+
+## Stepping and time settings (get from process info or solving info)
+#delta time
+delta_time = ProjectParameters["problem_data"]["time_step"].GetDouble()
+#start step
+step       = 0
+#start time
+time       = ProjectParameters["problem_data"]["start_time"].GetDouble()
+#end time
+end_time   = ProjectParameters["problem_data"]["end_time"].GetDouble()
+
+main_model_part.Nodes[10].SetSolutionStepValue(DISPLACEMENT_X, 0, 0.01)    
+main_model_part.Nodes[10].SetSolutionStepValue(DISPLACEMENT_Y, 0, 0.0) 
+main_model_part.Nodes[10].Fix(DISPLACEMENT_X) 
+main_model_part.Nodes[10].Fix(DISPLACEMENT_Y)
+
+
+main_model_part.Nodes[11].SetSolutionStepValue(DISPLACEMENT_X, 0, 0.01)    
+main_model_part.Nodes[11].SetSolutionStepValue(DISPLACEMENT_Y, 0, 0.1)
+main_model_part.Nodes[11].Fix(DISPLACEMENT_X) 
+main_model_part.Nodes[11].Fix(DISPLACEMENT_Y)
+
+
+main_model_part.Nodes[12].SetSolutionStepValue(DISPLACEMENT_X, 0, 0.01)    
+main_model_part.Nodes[12].SetSolutionStepValue(DISPLACEMENT_Y, 0, 0.01)
+main_model_part.Nodes[12].Fix(DISPLACEMENT_X) 
+main_model_part.Nodes[12].Fix(DISPLACEMENT_Y)
+
+constraintMaker = AddMultiPointConstraint() 
+
+# solving the problem (time integration)
+while(time <= end_time):
+
+    time = time + delta_time
+    step = step + 1
+    main_model_part.CloneTimeStep(time)
+
+   
+    # print process info
+    ## 
+    print("##########################")
+    print("Time :: ",time)
+    print("Step :: ",step)
+    
+    for process in list_of_processes:
+        process.ExecuteInitializeSolutionStep()
+
+    gid_output.ExecuteInitializeSolutionStep()
+    
+    print("########################## Solving START")
+    solver.Solve()
+    print("########################## Solving END")
+    
+    if(step == 1):
+        main_model_part.Nodes[6].SetValue(IS_SLAVE, True)
+        main_model_part.Nodes[7].SetValue(IS_SLAVE, True)
+        main_model_part.Nodes[9].SetValue(IS_SLAVE, True)
+        print("Setting MPC Constraints .. !!") 
+       
+        constraintMaker.ApplyConstraint(main_model_part.Nodes[16],DISPLACEMENT_Y, main_model_part.Nodes[6], DISPLACEMENT_Y,  1.0 )  
```

Comment:
they should be catalogues in submodelparts, this is exactly what submodelparts are about.

also note that you can have cases in which MPCs are defined once in the 

         ExecuteInitialize

or others that needs to be refreshed at every time step (think of slip BCs if the normal changes)
in this case you would need to do it in the 

       ExecuteInitializeSolutionStep